src/t8_forest/t8_forest_ghost.cxx

```diff
@@ -1929,4 +1880,348 @@ t8_forest_ghost_destroy (t8_forest_ghost_t *pghost)
+  num_elements_in_tree = t8_forest_get_tree_num_elements (forest, 0);
+
+  for (ielement = 0; ielement < num_elements_in_tree; ++ielement, ++current_index) {
+    element = t8_forest_get_element_in_tree (forest, 0, ielement);
+    /** compute the stencil elements */
+    add_stencil_to_ghost (forest, element, eclass_scheme, eclass_scheme->t8_element_level (element), tree_class, 0,
+                          ielement);
+  }
+
+  communicate_ghost_elements (forest);
+}
+
+/**
+ * Function to compute the owner of an element
+ * \param [in]      forest a commit uniform forest which is partitioned
+ * \param [in]      eclass_scheme should fit to the element
+ * \param [in]      element compute owner of this
+ * \param [out]     owner
+ * \note: the function use, that the linear id of the element is the same as the global index
+ * for example this is true for uniform meshes.
+ * the function also use, that the forest is partitioned.
+ * The owner is found in O(1)
+ */
+int
+t8_forest_ghost_interface_stencil_get_remote_rank (t8_forest_t forest, t8_eclass_scheme_c *eclass_scheme,
+                                                   t8_element_t *element)
+{
+  t8_gloidx_t global_num_elements = t8_forest_get_global_num_elements (forest);
+  t8_linearidx_t lin_id = eclass_scheme->t8_element_get_linear_id (element, eclass_scheme->t8_element_level (element));
+
+  t8_linearidx_t b_0 = global_num_elements / forest->mpisize;
+  t8_linearidx_t r = global_num_elements % forest->mpisize;
+
+  int p_guess = lin_id / b_0;
+  if (p_guess < 2 || r == 0) {
+    return p_guess;
+  }
+  int x = (p_guess * r) / forest->mpisize;
+
+  int owner = (lin_id - x) / b_0;
+
+  return owner;
+}
+
+void
+t8_forest_ghost_stencil::add_stencil_to_ghost (t8_forest_t forest, const t8_element_t *element,
+                                               t8_eclass_scheme_c *eclass_scheme, int level, t8_eclass_t tree_class,
+                                               t8_locidx_t ltreeid, t8_locidx_t ielement)
+{
+
+  t8_locidx_t iface_neighbor, ineighbor;
+  t8_element_t *face_neighbor;
+  t8_element_t *neighbor;
+  t8_eclass_t eclass = t8_forest_get_eclass (forest, 0);
+
+  /**
+   * First loop over the F elements, this are the face-neighbors of E
+   */
+  for (iface_neighbor = 0; iface_neighbor < eclass_scheme->t8_element_num_faces (element); ++iface_neighbor) {
+    eclass_scheme->t8_element_new (1, &face_neighbor);
+    eclass_scheme->t8_element_new (1, &neighbor);
+    /* Compute one F */
+    int neighbor_face;
+    int face_neighbor_exists
+      = eclass_scheme->t8_element_face_neighbor_inside (element, face_neighbor, iface_neighbor, &neighbor_face);
+    /* if the F exists */
+    if (face_neighbor_exists) {
+      /* compute the mpirank for F, and if its not owns by this process, add it to ghost */
+      int remote_rank = t8_forest_ghost_interface_stencil_get_remote_rank (forest, eclass_scheme, face_neighbor);
```

Comment:
```suggestion
      const int remote_rank = t8_forest_ghost_interface_stencil_get_remote_rank (forest, eclass_scheme, face_neighbor);
```

Reply:
I reuse the variable (for the remote rank of the neighbor of the neighbor), so I can not make it const.
If it is a better style I could use two different variables for the remote_rank of the elements. Than I use const.